ldk/go/loopClient.go

```diff
@@ -29,113 +30,69 @@ func (m *LoopClient) LoopStart(host Sidekick) error {
 		Impl: host.Clipboard(),
 	}
 
-	clipboardServerFunc := func(opts []grpc.ServerOption) *grpc.Server {
-		m.s = grpc.NewServer(opts...)
-		proto.RegisterClipboardServer(m.s, clipboardHostServer)
-		return m.s
-	}
-
-	clipboardBrokerID := m.broker.NextId()
-	go m.broker.AcceptAndServe(clipboardBrokerID, clipboardServerFunc)
-
 	//setup keyboard server
 	keyboardHostServer := &KeyboardServer{
 		Impl: host.Keyboard(),
 	}
 
-	keyboardServerFunc := func(opts []grpc.ServerOption) *grpc.Server {
-		m.s = grpc.NewServer(opts...)
-		proto.RegisterKeyboardServer(m.s, keyboardHostServer)
-		return m.s
-	}
-
-	keyboardBrokerID := m.broker.NextId()
-	go m.broker.AcceptAndServe(keyboardBrokerID, keyboardServerFunc)
-
 	processHostServer := &ProcessServer{
 		Impl: host.Process(),
 	}
 
-	processServerFunc := func(opts []grpc.ServerOption) *grpc.Server {
-		m.s = grpc.NewServer(opts...)
-		proto.RegisterProcessServer(m.s, processHostServer)
-		return m.s
-	}
-
-	processBrokerID := m.broker.NextId()
-	go m.broker.AcceptAndServe(processBrokerID, processServerFunc)
-
 	cursorHostServer := &CursorServer{
 		Impl: host.Cursor(),
 	}
 
-	cursorServerFunc := func(opts []grpc.ServerOption) *grpc.Server {
-		m.s = grpc.NewServer(opts...)
-		proto.RegisterCursorServer(m.s, cursorHostServer)
-		return m.s
-	}
-
-	cursorBrokerID := m.broker.NextId()
-	go m.broker.AcceptAndServe(cursorBrokerID, cursorServerFunc)
-
 	filesystemHostServer := &FilesystemServer{
 		Impl: host.Filesystem(),
 	}
 
-	filesystemServerFunc := func(opts []grpc.ServerOption) *grpc.Server {
+	brokerID := m.broker.NextId()
+
+	readyChan := make(chan bool)
+
+	serverFunc := func(opts []grpc.ServerOption) *grpc.Server {
 		m.s = grpc.NewServer(opts...)
 		proto.RegisterFilesystemServer(m.s, filesystemHostServer)
+		proto.RegisterCursorServer(m.s, cursorHostServer)
+		proto.RegisterProcessServer(m.s, processHostServer)
+		proto.RegisterKeyboardServer(m.s, keyboardHostServer)
+		proto.RegisterClipboardServer(m.s, clipboardHostServer)
+		proto.RegisterStorageServer(m.s, storageHostServer)
+		proto.RegisterWhisperServer(m.s, whisperHostServer)
+		readyChan <- true
 		return m.s
 	}
 
-	filesystemBrokerID := m.broker.NextId()
-	go m.broker.AcceptAndServe(filesystemBrokerID, filesystemServerFunc)
+	go m.broker.AcceptAndServe(brokerID, serverFunc)
+
+	<-readyChan
 
+	serviceHosts := &proto.ServiceHosts{
+		HostBrokerId: brokerID,
+	}
 	_, err := m.client.LoopStart(ctx, &proto.LoopStartRequest{
-		ServiceHosts: &proto.ServiceHosts{
-			HostStorage:    storageBrokerID,
-			HostWhisper:    whisperBrokerID,
-			HostClipboard:  clipboardBrokerID,
-			HostKeyboard:   keyboardBrokerID,
-			HostProcess:    processBrokerID,
-			HostCursor:     cursorBrokerID,
-			HostFilesystem: filesystemBrokerID,
-		},
+		ServiceHosts: serviceHosts,
 		// TODO: Define Session here
 		Session: &proto.Session{
 			LoopID: "LOOP_ID",
 			Token:  "TOKEN",
 		},
 	})
-
+	if err != nil {
+		fmt.Println("loopClient.go ERROR FOLLOWS")
```

Comment:
Seems like these 2 lines should be combined into 1.  Also, should this be using a logger?  I feel like stdout just gets eaten by Sidekick and not recorded anywhere.

Reply:
We should, but haven't yet, add loggers on the Olive Helps server sides. I'll make a card for that.

I don't think this should be one line, I'd like to have the first line separated from the second. These have been writing to stdout so it worked for me in the debugging process.

Reply:
SIDE-985 created

Reply:
@DrewRepaskyOlive I **think** OH still reads stdout and logs it, but it won't be nearly as nice as something that's using the LDK's `Logger`s, which format the messages in a better way for OH to parse. So I'd agree that using the LDK `Logger`s are the best option, but normal stdout _should_ still work, worst case scenario

Reply:
If we're running from the command-line, then devs can see stdout in the console.  I just meant that when I make a call to `fmt.Printf` inside of a Loop, those messages do not show up in Sidekick log _files_, so we have to use a logger instead.  I'm more concerned with the published desktop app run by end users and not losing error context there.
Thanks for opening that ticket @richseviora